boot/bootutil/src/encrypt_ec256.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2018-2019 JUUL Labs
+ * Copyright (c) 2019-2021 Arm Limited
+ */
+
+#include "mcuboot_config/mcuboot_config.h"
+
+#if defined(MCUBOOT_ENC_IMAGES) && defined(MCUBOOT_ENCRYPT_EC256)
+
+#include "encrypted_priv.h"
+
+static const uint8_t ec_pubkey_oid[] = MBEDTLS_OID_EC_ALG_UNRESTRICTED;
+static const uint8_t ec_secp256r1_oid[] = MBEDTLS_OID_EC_GRP_SECP256R1;
+
+/*
+ * Parses the output of `imgtool keygen`, which produces a PKCS#8 elliptic
+ * curve keypair. See RFC5208 and RFC5915.
+ */
+static int
+parse_ec256_enckey(uint8_t **p, uint8_t *end, uint8_t *private_key)
+{
+    int rc;
+    size_t len;
+    int version;
+    mbedtls_asn1_buf alg;
+    mbedtls_asn1_buf param;
+
+    if ((rc = mbedtls_asn1_get_tag(p, end, &len,
+                    MBEDTLS_ASN1_CONSTRUCTED | MBEDTLS_ASN1_SEQUENCE)) != 0) {
+        return -1;
```

Comment:
I think it can be a subject for an all other PR.
The error values of parse_ec256_enckey are hardcoded but they are also not used.
```C
    rc = parse_ec256_enckey(&cp, cpend, private_key);
    if (rc) {
        return -1;
    }
```
The EC256 configuration is not the only one affected by this problem.

It would be a good opportinunity to discuss if MCUBoot should define its own error values or just propagate the one
it receives from the crypto module it uses (for example : Mbedtls).

Reply:
> 
> It would be a good opportinunity to discuss if MCUBoot should define its own error values or just propagate the one it receives from the crypto module it uses (for example : Mbedtls).

The entire IT industry should figure it out, it is really annoying to be forced to debug where did the `-EINVAL` comes from, because everything returns it for some reason or an other.

Reply:
I checked how parse_ec256_enckey return value is handled on the main branch and the error is propagated when there is one.
So it is a change I made when I did the refactoring...
I reverted this.